File: CodingChallengeApi/CodingChallengeApi.Core/Models/Employee.cs
namespace CodingChallengeApi.Core.Models;
public class Employee : Person
{
    public int Id { get; set; }
    public static int BaseSalary => 2000;
    public decimal Salary => BaseSalary - Deductions;
    public decimal Deductions { get; set; }
    public IList<Person> Dependents { get; set; } = new List<Person>();
}

File: coding-challenge-webapp/src/app/models/employee.ts
import { Person } from "./person";

export class Employee extends Person {
    id: number;
    salary: number;
    dependents: Person[];
    deductions: number;

    constructor(params: any){
        super(params);
        this.id = params?.id ?? 0;
        this.salary = params?.salary ?? 0;
        this.dependents = Person.prototype.buildPersonArray(params?.dependents) ?? [];
        this.deductions = params?.deductions ?? 0;
    }

    buildEmployeeArray(params: any[]): Employee[] {
        const employees: Employee[] = [];
        params?.forEach(element => {
            employees.push(new Employee(element))
        });
        return employees;
    }
}